parser-typechecker/src/Unison/Parser.hs

```diff
@@ -60,13 +60,19 @@ instance Monoid UniqueName where
   mappend (UniqueName f) (UniqueName g) =
     UniqueName $ \pos len -> f pos len <|> g pos len
 
-uniqueBase58Namegen :: IO UniqueName
-uniqueBase58Namegen = do
-  rng <- Random.getSystemDRG
-  pure . UniqueName $ \pos lenInBase32Hex -> go pos lenInBase32Hex rng
+-- Should be used when we actually do want the result to be different every time
+-- (most cases)
+-- But we actually want it to be deterministic when running transcripts, 
+-- in which case we need to be able to control the random seed. If that is the case you should use 
+-- `uniqueBase32Namegen`
```

Comment:
I didn’t quite follow this comment, does it have a mistake, or could you help me understand?